File: JavaScript/Day 03/day3-part2/day3-part2.js
const fs = require("fs");
const data = fs
  .readFileSync("input.in", { encoding: "utf8", flag: "r" })
  .toString()
  .split("\r\n")
  .map(String);

const countTrees = (right, down) => {
  let x = 0,
    y = 0,
    countTrees = 0;
  const rows = data.length,
    columns = data[0].length;

  while (x < rows) {
    if (data[x][y] == "#") {
      countTrees++;
    }
    x = x + down;
    y = (y + right) % columns;
  }

  return countTrees;
};

let multiply = countTrees(1, 1);
multiply *= countTrees(3, 1);
multiply *= countTrees(5, 1);
multiply *= countTrees(7, 1);
multiply *= countTrees(1, 2);

fs.writeFileSync("output.out", multiply.toString());

File: C++/Day 03/day3-part2/day3-part2.cpp
#include "../../AOCHeaders/stdafx.h"


int countTrees(const std::vector<std::string>& map, const int right, const int down)
{
	int x{ 0 }, y{ 0 }, rows{ (int)map.size() }, columns{ (int)map[0].size() }, countTrees{ 0 };

	while (x < rows)
	{
		if (map[x][y] == '#')
		{
			countTrees++;
		}
		x = x + down;
		y = (y + right) % columns;
	}

	return countTrees;
}


int main()
{
	std::fstream in("input.in", std::fstream::in);
	std::fstream out("output.out", std::fstream::out);

	std::vector<std::string> map(std::istream_iterator<std::string>{in}, {});
	long long multiply{ 1 };

	multiply *= countTrees(map, 1, 1);
	multiply *= countTrees(map, 3, 1);
	multiply *= countTrees(map, 5, 1);
	multiply *= countTrees(map, 7, 1);
	multiply *= countTrees(map, 1, 2);

	out << multiply;

	in.close();
	out.close();
}